cloud/aws/bin/dumpdb.py

```diff
@@ -0,0 +1,169 @@
+import os
+import ipaddress
+import sys
+import tempfile
+import shlex
+import subprocess
+from time import sleep
+from datetime import datetime
+import urllib.request
+
+from cloud.aws.templates.aws_oidc.bin import resources
+from cloud.aws.templates.aws_oidc.bin.aws_cli import AwsCli
+from cloud.shared.bin.lib import terraform
+from cloud.shared.bin.lib.config_loader import ConfigLoader
+from cloud.shared.bin.lib.print import print
+
+
+def run(config: ConfigLoader):
+    aws = AwsCli(config)
+    print(
+        "\n"
+        "\033[31m!!! WARNING !!!: This command will create a dump of the entire database, including personally identifiable information (PII). Ensure you take the utmost care in handling this data and store it in a secure location.\033[0m\n"
+        "\n"
+        'This process will set up a temporary EC2 host with access to the database, use SSH to run the pg_dump command on that host, then SCP the file to this machine. '
+        'You will need to confirm the application of the Terraform manifest that creates these temporary resources, and then confirm the teardown of these resources. '
+        'If something goes wrong and this process is interrupted before it tears down the resources, you can find them all with the "Module = dbaccess" tag in the AWS console. They should be deleted manually.'
+        "\n\n"
+        'The "ssh" and "ssh-keygen" commands must be available on your machine, typically provided by the openssh-client package. If you do not have these commands, you will need to install them before proceeding.'
+        "\n\n")
+    answer = input('Do you understand the risks and wish to proceed? (y/N): ')
+    if answer.lower() != 'y':
+        print('Exiting.')
+        return
+
+    timestamp = datetime.now().strftime('%Y-%m-%d_%H-%M-%S')
+
+    # Current working dir should be the 'checkout' folder, so go one level above.
+    default_file = os.path.join(
```

Comment:
**suggestion:** This is being set, but it doesn't seem like we're actually using it when storing the file? I think we should use this naming on both the EC2 and local filesystem